pkg/k8s/pipelineRun.go

```diff
@@ -208,72 +212,74 @@ func (r *pipelineRun) String() string {
 }
 
 // UpdateResult of the pipeline run
-func (r *pipelineRun) UpdateResult(result api.Result) error {
+func (r *pipelineRun) UpdateResult(result api.Result) {
 	r.ensureCopy()
-	return r.changeStatusAndUpdateSafely(func() error {
-		r.apiObj.Status.Result = result
+	// the call below returns the error from the function. This error is always nil. Hence there
+	// is no need to return that always-nil-error.
+	panicOnError(r.changeStatusAndStoreForRetry(func(s *api.PipelineStatus) (commitRecorderFunc, error) {
+		s.Result = result
 		now := metav1.Now()
-		r.apiObj.Status.FinishedAt = &now
-		return nil
-	})
+		s.FinishedAt = &now
+		return nil, nil
+	}))
 }
 
 // UpdateContainer ...
-func (r *pipelineRun) UpdateContainer(c *corev1.ContainerState) error {
+func (r *pipelineRun) UpdateContainer(c *corev1.ContainerState) {
 	if c == nil {
-		return nil
+		return
 	}
 	r.ensureCopy()
-	return r.changeStatusAndUpdateSafely(func() error {
-		r.apiObj.Status.Container = *c
-		return nil
-	})
+	panicOnError(r.changeStatusAndStoreForRetry(func(s *api.PipelineStatus) (commitRecorderFunc, error) {
```

Comment:
```suggestion
	r.mustChangeStatusAndStoreForRetry(func(s *api.PipelineStatus) (commitRecorderFunc, error) {
```
May be you should use the must pattern as in https://pkg.go.dev/regexp#MustCompile

func mustChangeStatusAndStoreForRetry(change changeFunc) {
       err := r.changeStatusAndStoreForRetry
	if err != nil {
		panic(err)
	}
}

Reply:
done